Pool/ScoreTheme.vb

```diff
@@ -73,89 +48,87 @@ Public Class ScoreTheme
-            If Not player.Value.Equals("Error") Then
-                cBox.Items.Add(player.Value)
-            End If
-        Next
-    End Sub
-    Public Sub FillCBoxAll(ByVal allList As Hashtable, ByRef cBox As ComboBox, ByRef lblError As Label)
+
+    ''' <summary>
+    ''' Fill combox from hashtable
+    ''' </summary>
+    ''' <param name="cBox"></param>
+    ''' <param name="allDT"></param>
+    Public Shared Sub FillBoxfromHT(ByRef cBox As ComboBox, ByVal allDT As Hashtable)
         cBox.Items.Clear()
-        For Each name As DictionaryEntry In allList
-            If Screen = 0 AndAlso name.Value.ToString.Contains("Error") Then
-                Screen = -1
-                lblError.Text = "Error getting all players"
-                Exit For
-            Else
-                cBox.Items.Add(name.Value)
-                Screen = AppState.SelectPlayer
+        cBox.Items.Add("Choose")
+        For Each entry As DictionaryEntry In allDT
+            If Not entry.Value.Equals("Error") Then
+                cBox.Items.Add(entry.Value)
             End If
         Next
     End Sub
 
-    Public Function ValidateCBox(ByRef cBox As ComboBox) As Boolean
-        If cBox.SelectedItem IsNot Nothing AndAlso Not String.IsNullOrEmpty(cBox.SelectedItem.ToString) Then
+    'Public Sub FillCBoxAll(ByVal allList As Hashtable, ByRef cBox As ComboBox, ByRef lblError As Label)
+    '    cBox.Items.Clear()
+    '    For Each name As DictionaryEntry In allList
+    '        If Screen = 0 AndAlso name.Value.ToString.Contains("Error") Then
+    '            Screen = -1
+    '            lblError.Text = "Error getting all players"
+    '            Exit For
+    '        Else
+    '            cBox.Items.Add(name.Value)
+    '            Screen = AppState.SelectPlayer
+    '        End If
+    '    Next
+    'End Sub
+
+    ''' <summary>
+    ''' Use to validate selected item on cBox
+    ''' </summary>
+    ''' <param name="cBox"></param>
+    ''' <returns></returns>
+    Public Shared Function ValidateCBox(ByRef cBox As ComboBox) As Boolean
+        If cBox.SelectedItem IsNot Nothing AndAlso Not String.IsNullOrEmpty(cBox.SelectedItem.ToString) AndAlso Not cBox.SelectedItem.ToString.Contains("Choose") Then
             Return True
         Else
             Return False
         End If
     End Function
 
-    Public Function getSubs(ByVal gameMode As Games) As List(Of String)
+    ''' <summary>
+    ''' Get missing required fields to display
+    ''' </summary>
+    ''' <param name="controls"></param>
+    ''' <returns></returns>
+    Public Shared Function GetMissingFieldNames(ByRef controls As Control()) As String
+        Dim missingFields As New StringBuilder
+        For Each field In controls
+            If String.IsNullOrEmpty(field.Text) Then
+                missingFields.Append(field.Name.Substring(3) & ",")
+            End If
+        Next
+        Return missingFields.ToString.Trim(",")
+    End Function
```

Comment:
We shouldn't use the control name if we present it to the user. If we are using this in enough spots then we should change it to use a more clear cut string. Remove it completely if possible and stick to the alert boxes.

Reply:
This was only used in 2 spots and was removed to use the alert dialog.